### ISSUE TITLE
Rustc (nightly) panic!s for this incorrect pattern

### ISSUE THREAD

**Issue by 0x53ee71ebe11e** [rust-lang/rust#39387]
(I edited this issue, because it didn't have anything to do with the macro I originally tried)

I tried this code:
```rust
fn main() {
    let x;
    (x) = (1);
    println!("x={}",x);
}
```

I expected the compiler to give a reasonable error message about `(x) = (1)` being incorrect (Is it really incorrect though? I'm not completely sure about that).

But I got this:
```
error: internal compiler error: unexpected panic

note: the compiler unexpectedly panicked. this is a bug.

note: we would appreciate a bug report: https://github.com/rust-lang/rust/blob/master/CONTRIBUTING.md#bug-reports

thread 'rustc' panicked at 'internal error: entered unreachable code', /buildslave/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libsyntax/parse/parser.rs:2738
note: Run with `RUST_BACKTRACE=1` for a backtrace.
```

```
$ rustc --version
rustc 1.16.0-nightly (df8debf6d 2017-01-25)
```

**Comment by 0x53ee71ebe11e** [comment]
Meh, this is nonsense, I'll try to figure out where exactly it panics and then I'll write a new issue about it.
